src/PowerShell/AzPSScriptBuilder.ts

```diff
@@ -0,0 +1,111 @@
+import AzPSConstants from "./AzPSConstants";
+import { LoginConfig } from '../common/LoginConfig';
+
+export default class AzPSScriptBuilder {
+
+    static getImportLatestModuleScript(moduleName: string): string {
+        let script = `try {
+            $ErrorActionPreference = "Stop"
+            $WarningPreference = "SilentlyContinue"
+            $output = @{}
+            $latestModulePath = (Get-Module -Name '${moduleName}' -ListAvailable | Sort-Object Version -Descending | Select-Object -First 1).Path
+            Import-Module -Name $latestModulePath
+            $output['${AzPSConstants.Result}'] = $latestModulePath
+            $output['${AzPSConstants.Success}'] = "true"
+        }
+        catch {
+            $output['${AzPSConstants.Error}'] = $_.exception.Message
+        }
+        return ConvertTo-Json $output`;
+
+        return script;
+    }
+
+    static async getAzPSLoginScript(loginConfig: LoginConfig) {
+        let loginMethodName = "";
+        let commands = 'Clear-AzContext -Scope Process; ';
+        commands += 'Clear-AzContext -Scope CurrentUser -Force -ErrorAction SilentlyContinue; ';
+
+        if (loginConfig.environment.toLowerCase() == "azurestack") {
+            commands += `Add-AzEnvironment -Name '${loginConfig.environment}' -ARMEndpoint '${loginConfig.resourceManagerEndpointUrl}' | out-null;`;
+        }
+        if (loginConfig.authType === LoginConfig.AUTH_TYPE_SERVICE_PRINCIPAL) {
+            if (loginConfig.servicePrincipalKey) {
```

Comment:
It seems the nested if-else can be refactored to something like `switch (loginConfig.GetLoginFlow()) { ... }` but I don't have a strong opinion here.